### ISSUE TITLE
osgEarthQT trunk doesn't compile with QT5

### ISSUE THREAD

**Issue by remoe** [gwaldron/osgearth#464]
I have the following compile issue when I build the current trunk with QT 5.2:

``` log
moc_AnnotationDialogs.cpp(53): error C2653: 'osgEarth' : is not a class or namespace name
moc_CollapsiblePairWidget.cpp(67): error C2653: 'osgEarth' : is not a class or namespace name
moc_LayerManagerWidget.cpp(70): error C2653: 'osgEarth' : is not a class or namespace name
moc_DataManager.cpp(75): error C2653: 'osgEarth' : is not a class or namespace name
moc_AnnotationDialogs.cpp(61): error C2653: 'osgEarth' : is not a class or namespace name
moc_AnnotationListWidget.cpp(90): error C2653: 'osgEarth' : is not a class or namespace name
```

UPDATE: I found that the following entry is missed in moc_AnnotationDialogs.cpp:
# include "../../../src/osgEarthQt/AnnotationDialogs"

I use the following environment:

QT5.2 on windows VS2013.1 x64

UPDATE2:
I'm not sure if this has to do with:
http://forum.openscenegraph.org/viewtopic.php?t=13213

**Comment by DomTomCat** [comment]
Problem is the same for Qt5.4 (VS2010, osgearth 2.6, cmake 3.0.2).

@NikoMoreno ntt73's solution is only a quick fix to finish compiling, doesn't solve the source of the problem. 
Say you're building in osgearth-osgearth-2.6\build\
then cd to osgearth-osgearth-2.6\build\src\osgEarthQt and use a batch file like this:

set dir=..\..\..\src\osgEarthQt
moc -f %dir%\ViewerWidget @.\moc_ViewerWidget.cpp_parameters
moc -f %dir%\AnnotationDialogs @.\moc_AnnotationDialogs.cpp_parameters
moc -f %dir%\AnnotationListWidget @.\moc_AnnotationListWidget.cpp_parameters
moc -f %dir%\AnnotationToolbar @.\moc_AnnotationToolbar.cpp_parameters
moc -f %dir%\CollapsiblePairWidget @.\moc_CollapsiblePairWidget.cpp_parameters
moc -f %dir%\DataManager @.\moc_DataManager.cpp_parameters
moc -f %dir%\LayerManagerWidget @.\moc_LayerManagerWidget.cpp_parameters
moc -f %dir%\LOSControlWidget @.\moc_LOSControlWidget.cpp_parameters
moc -f %dir%\LOSCreationDialog @.\moc_LOSCreationDialog.cpp_parameters
moc -f %dir%\MapCatalogWidget @.\moc_MapCatalogWidget.cpp_parameters
moc -f %dir%\TerrainProfileGraph @.\moc_TerrainProfileGraph.cpp_parameters
moc -f %dir%\TerrainProfileWidget @.\moc_TerrainProfileWidget.cpp_parameters

then finish building. Obviously this can be much nicer but it does the trick.

Btw, same procedure, if you want to finish building the osgearth qt samples, as well

**Comment by smilewithyou02** [comment]
Why this issue closed? I still encounter it in Qt5.5+osg3.4+osgearth2.7. Is there a good way to fix it?

**Comment by emminizer** [comment]
I am building Qt 5.4 and 5.5, OSG 3.2.3, and the master of osgEarth, using CMake 3.1-3.3, without problem.  I do specify CMAKE_PREFIX_PATH, Qt5Core_DIR, Qt5Widgets_Dir, Qt5Gui_DIR, and Qt5OpenGL_DIR on the configure line.

I needed to do similar flags to get OSG building.

This is with MSVC 2010 through 2015, and g++ 4.1 and 4.4.  We've been building osgEarth against Qt5 since about April 2014, Qt 5.2.  At that point we had to modify MOC to get it to play nicely with extensionless headers, but again that hit OSG too.  Later versions of Qt fixed that bug.
